src/com/sun/jna/Memory.java

```diff
@@ -862,4 +876,25 @@ protected static long malloc(long size) {
     public String dump() {
         return dump(0, (int)size());
     }
+
+    /**
+     * Check whether the supplied Pointer object points into the memory region
+     * backed by this memory object. The intention is to prevent premature GC
+     * of the Memory object.
+     *
+     * @param target Pointer to check
+     * @return {@code target} if target does not point into the region covered
+     * by this memory object, a newly {@code SharedMemory} object, if the pointer
+     * points to memory backed by this Memory object.
+     */
+    private Pointer shareIfInternalPointer(Pointer target) {
+        if(target == null) {
+            return null;
+        }
+        if (target.peer >= this.peer && (this.peer + this.size) > target.peer) {
```

Comment:
(Preference/Style) I think this would be more clear if the second conditional order was reversed (`target.peer >= foo && target.peer < bar`).